content/articles/2019-09-tidyr-1-0-0.Rmarkdown

```diff
@@ -0,0 +1,316 @@
+* `hoist()` is similar to `unnest_wider()` but only plucks out selected
+  components, and can reach down multiple levels.
+
+To see them in action, take this small sample of `repurrrsive::got_chars`. It contains data about three characters from the Game of Thrones:
+
+```{r}
+characters <- list(
+  list(
+    name = "Theon Greyjoy",
+    aliases = c("Prince of Fools", "Theon Turncloak", "Theon Kinslayer"),
+    alive = TRUE
+  ),
+  list(
+    name = "Tyrion Lannister",
+    aliases = c("The Imp", "Halfman", "Giant of Lannister"),
+    alive = TRUE
+  ),
+  list(
+    name = "Arys Oakheart",
+    alive = FALSE
+  )
+)
+```
+
+To work with the new tidyr rectangling tools, we first put the list into a data frame, creating a list-column:
+
+```{r}
+got <- tibble(character = characters)
+got
+```
+
+We can then use `unnest_wider()` to make each element of that list into a column:
+
+```{r}
+got %>% 
+  unnest_wider(character)
+```
+
+Followed by `unnest_longer()` to turn each alias into its own row:
+
+```{r}
+got %>% 
+  unnest_wider(character) %>% 
+  unnest_longer(aliases)
+```
+
+Even more conveniently, you can use `unnest_auto()` to guess which direction a list column should be unnested in. Here it yields the same results as above, and the messages tell you why:
+
+```{r, results = FALSE}
+got %>% 
+  unnest_auto(character) %>% 
+  unnest_auto(aliases)
+```
+
+Alternatively, you can use `hoist()` to reach deeply into a data structure and put out just the pieces you need:
+
+```{r}
+got %>% hoist(character, 
+  name = "name",
+  alias = list("aliases", 1),
+  alive = "alive"
+)
+```
+
+This replaces our previous recommendation to use `dplyr::mutate()` and `purrr::map()` together:
+
+```{r, results = FALSE}
+got %>% mutate(
+  name = purrr::map_chr(character, "name"),
+  alias = purrr::map_chr(character, list("aliases", 1), .default = NA),
+  alive = purrr::map_lgl(character, "alive")
+)
+```
+
+Learn more in [`vignette("rectangle")`](http://tidyr.tidyverse.org/dev/articles/rectangle.html).
+
+## Nesting
+
+`nest()` and `unnest()` have been updated with new interfaces that are more closely aligned to evolving tidyverse conventions. The biggest change is to their operation with multiple columns: 
+
+```{R, eval = FALSE}
+# old
+df %>% nest(x, y, z)
+# new
+df %>% nest(data = c(x, y, z))
+
+# old
+df %>% unnest(x, y, z)
+# new
+df %>% unnest(c(x, y, z))
+```
+
+I've done my best to ensure that common uses of `nest()` and `unnest()` will continue to work, generating an informative warning telling you precisely how you need to update your code. Please [file an issue](https://github.com/tidyverse/tidyr/issues/new) if I've missed an important use case.
+
+Behind the scenes, I discovered that nesting (and unnesting) can be decomposed into the combination of two simpler operations:
+
+* `pack()` and `unpack()` pack and unpack multiple columns into a single data 
+  frame columns. 
+  
+* `chop()` and `unchop()` chop up rows into sets of list-columns. 
```

Comment:
```suggestion
* `chop()` rows of a column into a list-column and `unchop()` a list-column into rows.
```

Reply:
Consider: tie to `dplyr::group_by()`? In my mind, `group_by()` inserts "dotted lines" whereas as `chop()` inserts solid lines, if that makes any sense.

Reply:
Added a little verbiage; probably not terribly clear but hopefully it will get interested people thinking in the right direction.